### PR TITLE
fix(oas2): handle invalid server host

### PR DESCRIPTION
**Pull request by P0lip**
Addresses https://github.com/stoplightio/studio-internal/issues/1363

### CODE REVIEW

File: src/oas2/transformers/__tests__/servers.test.ts [hunk 1]
```diff
@@ -94,4 +99,21 @@ describe('translateToServers', () => {
       },
     ]);
   });
+
+  test('should handle invalid server host gracefully', () => {
+    expect(translateToServers({ host: 123 as any, basePath: '/base-path' }, { schemes: ['http', 'https'] })).toEqual(
```
Comment (lottamus) [inline src/oas2/transformers/__tests__/servers.test.ts]: Why not handle it as a string?

```ts
[
  {
    url: 'http://123'
  }
]
```

Comment (marbemac) [reply src/oas2/transformers/__tests__/servers.test.ts]: I think OK to just toss it - that would be a very odd server host ha.

File: src/oas2/transformers/servers.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ export function translateToServers(spec: Partial<Spec>, operation: Partial<Opera
       .scheme(scheme)
       .host(host);
 
-    if (basePath) {
-      uri = uri.path(basePath);
+    if (typeof basePath === 'string' && basePath.length > 0) {
+      uri = uri.path(String(basePath));
```
Comment (marbemac) [inline src/oas2/transformers/servers.ts]: If `basePath` is already a string, why the cast here?

File: src/oas2/transformers/__tests__/servers.test.ts (hunk 1 above)
Comment (lottamus) [reply src/oas2/transformers/__tests__/servers.test.ts]: Users do very _odd_ things with their OpenAPI and not much extra effort to support it lol 🤷‍♂

Comment (XVincentX) [reply src/oas2/transformers/__tests__/servers.test.ts]: From a Prism perspective, we do not use servers anyway so whatever works for you will work for us as well.

File: src/oas2/transformers/servers.ts (hunk 2 above)
Comment (P0lip) [reply src/oas2/transformers/servers.ts]:
```suggestion
      uri = uri.path(basePath);
```
Yeah, it's a leftover, as I hadn't had the check before I added it.

**Comment by marbemac** [comment]
Good to go @P0lip!

**Pull request merged**